### PR TITLE
Limit Netty Workers in NativeRealmIntegTestCase

### LINKED ISSUE #46803: NativeRealmIntegTests tests failed with OutOfMemoryError Direct buffer memory
All the tests in `NativeRealmIntegTests` failed today on `7.x` branch:
https://gradle-enterprise.elastic.co/s/q3qkmolurbooa

Something went wrong when allocating direct buffer memory in the Transport layer:
```java
WARNING: Uncaught exception in thread: Thread[Thread-51,5,TGRP-NativeRealmIntegTests]
java.lang.OutOfMemoryError: Direct buffer memory
	at __randomizedtesting.SeedInfo.seed([7168F60F3A169301]:0)
	at java.nio.Bits.reserveMemory(Bits.java:694)
	at java.nio.DirectByteBuffer.<init>(DirectByteBuffer.java:123)
	at java.nio.ByteBuffer.allocateDirect(ByteBuffer.java:311)
	at org.elasticsearch.transport.CopyBytesSocketChannel.lambda$static$0(CopyBytesSocketChannel.java:65)
	at java.lang.ThreadLocal$SuppliedThreadLocal.initialValue(ThreadLocal.java:284)
	at java.lang.ThreadLocal.setInitialValue(ThreadLocal.java:180)
	at java.lang.ThreadLocal.get(ThreadLocal.java:170)
	at org.elasticsearch.transport.CopyBytesSocketChannel.getIoBuffer(CopyBytesSocketChannel.java:141)
	at org.elasticsearch.transport.CopyBytesSocketChannel.doReadBytes(CopyBytesSocketChannel.java:121)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:148)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:597)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:551)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at java.lang.Thread.run(Thread.java:748)
```

It does not reproduce locally:

```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.authc.esnative.NativeRealmIntegTests.testUpdatingUserAndAuthentication" -Dtests.seed=7168F60F3A169301 -Dtests.security.manager=true -Dtests.locale=mk-MK -Dtests.timezone=Asia/Damascus -Dcompiler.java=12 -Druntime.java=8
```

### PR DESCRIPTION
**Pull request by original-brownbear**
The fact that this test randomly uses a relatively large number
of nodes and hence Netty worker threads created a problem with
running out of direct memory on CI.
Tests run with 512M heap (and hence 512M direct memory) by default.
On a CI worker with 16 cores, this means Netty will by default set
up 32 transport workers. If we get unlucky and a lot of them
actually do work (and thus instantiate a `CopyBytesSocketChannel`
which costs 1M per thread for the thread-local IO buffer) we
would run out of memory.

This specific failure was only seen with `NativeRealmIntegTests` so I
only added the constraint on the Netty worker count here.
We can add it to other tests (or `SecurityIntegTestCase`) if need be
but for now it doesn't seem necessary so I opted for least impact.

Closes #46803

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/test/NativeRealmIntegTestCase.java [hunk 1]
```diff
@@ -68,6 +70,16 @@ protected boolean addMockHttpTransport() {
         return templates;
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        Settings.Builder builder = Settings.builder().put(super.nodeSettings(nodeOrdinal));
+        // we are randomly running a large number of nodes in these tests so we limit the number of worker threads
+        // since the default of 2 * CPU count might use up too much direct memory for thread-local direct buffers for each node's
+        // transport threads
+        builder.put(Netty4Transport.WORKER_COUNT.getKey(), random().nextInt(3) + 1);
```
Comment (original-brownbear) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/test/NativeRealmIntegTestCase.java]: We use the same randomization for worker count in other tests so I picked it here as well ... no special reason for the parameter choice other than that.

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks @tbrooks8

**Comment by romseygeek** [comment]
#46803 just occurred in the 7.4 tests as well, can this fix be backported to that branch as well?

**Comment by original-brownbear** [comment]
This fix wasn't powerful enough it turns out, this also failed again on `7.x` that has this fix. I created a "stronger" fix in https://github.com/elastic/elasticsearch/pull/48267 that we can backport if/when it gets merged.

**Pull request merged**